source/anti-patterns/language.rst

```diff
@@ -9,10 +9,46 @@ Nix syntax supports URLs as verbatim, so one can write ``https://example.com`` i
 
 There's was an `RFC 45 <https://github.com/NixOS/rfcs/pull/45>`_ accepted to deprecate verbatim URLS and provides
 a number of arguments this features does more harm than good.
+ 
 
+``rec { ... }`` expression
+--------------------------
 
-``with`` expression
--------------------
+``rec`` allows you to reference variables within an attribute set.
+
+A simple example:
+
+.. code:: nix
+
+  rec {
+    a = 1;
+    b = a + 2;
+  }
+
+evaluating to ``{ a = 1; b = 3; }``.
+
+``b`` refers to ``a`` as ``rec`` makes all keys available within.
+
+There are a couple of pitfals:
+
+- it's possible to introduce recursion with the wrong shadowing, resulting in a weird error message
+
+- combining with overriding logic such as `overrideAttrs` function in nixpkgs it has suprising behavour
```

Comment:
"surprising behavior" is a very fuzzy term here, can we find at least a rough description of what the bad things™ are that might happen?

Reply:
Same here.